electrum/plugins/jade/jade.py

```diff
@@ -0,0 +1,464 @@
+    MIN_SUPPORTED_FW_VERSION = (0, 1, 32)
+
+    # For testing with qemu simulator (experimental)
+    SIMULATOR_PATH = None  # 'tcp:127.0.0.1:2222'
+    SIMULATOR_TEST_SEED = None  # bytes.fromhex('b90e532426d0dc20fffe01037048c018e940300038b165c211915c672e07762c')
+
+    def enumerate_serial(self):
+        # Jade is not really an HID device, it shows as a serial/com port device.
+        # Scan com ports looking for the relevant vid and pid, and use 'path' to
+        # hold the path to the serial port device, eg. /dev/ttyUSB0
+        devices = []
+        for devinfo in list_ports.comports():
+            device_product_key = (devinfo.vid, devinfo.pid)
+            if device_product_key in self.DEVICE_IDS:
+                device = Device(path=devinfo.device,
+                                interface_number=-1,
+                                id_=devinfo.serial_number,
+                                product_key=device_product_key,
+                                usage_page=-1,
+                                transport_ui_string=devinfo.device)
+                devices.append(device)
+
+        # Maybe look for Jade Qemu simulator if the vars are set (experimental)
+        if self.SIMULATOR_PATH is not None and self.SIMULATOR_TEST_SEED is not None:
+            try:
+                # If we can connect to a simulator and poke a seed in, add that too
+                client = Jade_Client(self.SIMULATOR_PATH, plugin=self)
+                device = Device(path=self.SIMULATOR_PATH,
+                                interface_number=-1,
+                                id_='Jade Qemu Simulator',
+                                product_key=self.DEVICE_IDS[0],
+                                usage_page=-1,
+                                transport_ui_string='simulator')
+                if client.jade.set_seed(self.SIMULATOR_TEST_SEED):
+                    devices.append(device)
+                client.close()
+            except Exception as e:
+                # If we get any sort of error do not add the simulator
+                _logger.debug("Failed to connect to Jade simulator at {}".format(self.SIMULATOR_PATH))
+                _logger.debug(e)
+
+        return devices
+
+    def __init__(self, parent, config, name):
+        HW_PluginBase.__init__(self, parent, config, name)
+
+        self.libraries_available = self.check_libraries_available()
+        if not self.libraries_available:
+            return
+
+        # Register our own serial/com port scanning function
+        self.device_manager().register_enumerate_func(self.enumerate_serial)
+
+    def get_library_version(self):
+        try:
+            from . import jadepy
+            version = jadepy.__version__
+        except ImportError:
+            raise
+        except:
+            version = "unknown"
+        return version
+
+    @runs_in_hwd_thread
+    def create_client(self, device, handler):
+        if handler:
+            self.handler = handler
```

Comment:
this is copied-over dead code, see https://github.com/spesmilo/electrum/commit/30623c352907d80dab421cbfb3dabf9c3847ac57

Reply:
Cheers, removed.